### PR TITLE
connmgr: Add func to iterate conn reqs.

### PR DESCRIPTION
**Pull request by davecgh**
**This is rebased on #2561**.

This adds a new func named `ForEachConnReq` to the connection manager which accepts a closure to call with each connection req known to the connection manager, including pending ones.

A test is also added to ensure proper functionality.

### CODE REVIEW

File: connmgr/connmanager.go [hunk 1]
```diff
@@ -200,6 +200,13 @@ type handleCancelPending struct {
 	done chan error
 }
 
+// handleForEachConnReq is used to iteratoe all known connection requests to
```
Comment (dnldd) [inline connmgr/connmanager.go:203]: s / iteratoe / iterate

File: connmgr/connmanager.go [hunk 2]
```diff
@@ -598,6 +625,30 @@ func (cm *ConnManager) CancelPending(addr net.Addr) error {
...
 }
 
+// ForEachConnReq calls the provided function with each connection request known
+// to the connection manager, including pending requests.  Returning an error
+// from the provided function will stop the iteration early and return said
+// error from this function.
+//
+// This function is safe for concurrent access.
+//
+// NOTE: This must note call any other connection manager methods during
```
Comment (dnldd) [inline connmgr/connmanager.go:635]: s / note / not

File: connmgr/connmanager.go (hunk 1 above)
Comment (davecgh) [reply connmgr/connmanager.go:203]: Fixed.

File: connmgr/connmanager.go (hunk 2 above)
Comment (davecgh) [reply connmgr/connmanager.go:635]: Changed.

**Review by dnldd — APPROVED** [review]

**Review by rstaudt2 — APPROVED** [review]

File: connmgr/connmanager_test.go [hunk 3]
```diff
@@ -803,6 +803,113 @@ func TestConnectContext(t *testing.T) {
...
+		},
+	}
+	go cmgr.Connect(context.Background(), cr)
+	select {
+	case <-pending:
+	case <-time.After(time.Millisecond * 5):
+		t.Fatal("timeout waiting for pending connection")
+	}
+
+	// Ensure the expected num
```
Comment (davecgh) [inline connmgr/connmanager_test.go:883]: Looks like I forgot to finish this comment.

Comment (davecgh) [reply connmgr/connmanager_test.go:883]: Fixed.

**Pull request merged**